### DISCUSSION TITLE
What are the limitations of PowerShell 5.1 version?

### DISCUSSION THREAD

**Discussion by chenxizhang** [chenxizhang/openai-powershell#201] [Use cases]
PowerShell 5.1 is the default version that comes with Windows starting from Windows 7. It can be quickly launched using the command `powershell`, and it features a blue icon.

![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/6b19991f-9502-4914-8ca7-e9c989f6ddf6)

It is the last version within the major release of PowerShell 5, followed by PowerShell 6 and 7, also known as PowerShell Core. The fundamental difference between them is that PowerShell 5.1 is developed based on the .NET Framework 4.0 version.

![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/4d8b5cce-70e3-4c1c-bb07-10737ab002aa)

On the other hand, PowerShell Core is developed based on .NET Core.

![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/21a223ee-9a0a-4104-b06f-9eeb953610a2)

As such, PowerShell 5.1 can only run on Windows, hence it is also known as Windows PowerShell. PowerShell Core, however, can run cross-platform, such as on macOS and Linux without any issues, and it is called Microsoft PowerShell.

So, regarding this specific module, are there any limitations placed on version 5.1?

**No**! One of our goals when developing this module was to ensure compatibility with 5.1. The advantage of this approach is that Windows users already have it installed by default, eliminating the need for any additional installation steps. For this purpose, careful compatibility handling was implemented in the code, such as support for UTF-8, for which a lot of additional code was written.

There is only one feature experience that differs, which is that since 5.1 does not support streaming, when you use the `chat` command to create a desktop version of ChatGPT, you won't see the typewriter effect. However, this does not impact usability.

![GIF 5-6-2024 5-06-22 PM](https://github.com/chenxizhang/openai-powershell/assets/1996954/933ed553-a19f-4ab5-95b8-2c9c3dfa48af)

**Comment by xiaoyangliu1123** [comment]
When I use PowerShell 5.1, the result from Kimi and Zhipu contains lots of "????" string (but Azure Open AI works well for both English and Chinese). The problem cannot be solved by changing the encoding format of powershell. 

This issue doesn't appear in Powershell 7.

**Reply by chenxizhang** [reply]
reproduced this issue, it seems like related to the model itself (kimi), since it works well if I use Azure OpenAI. Will take a look into it with lower priority.

![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/4321fa22-bf67-41ba-b2ca-3ee1e3b416fa)


![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/d3713d82-d42b-491d-9322-514c52dbd7fd)

**Reply by chenxizhang** [reply]
@xiaoyangliu1123 The bug (#206 ) has been fixed. Please update the module by using `update-Module code365scripts.openai` command, and have a try. 

![image](https://github.com/chenxizhang/openai-powershell/assets/1996954/1c4c6f92-98c9-49e6-b5eb-aa486cd70b05)

**Reply by xiaoyangliu1123** [reply]
The new version works for me! No issue now, thanks!
